### ISSUE TITLE
Create a stripped down tokenlist for tokens relevant for trading.

### ISSUE THREAD

**Issue by TomAFrench** [element-fi/elf-tokenlist#7]
Currently the tokenlist includes all tokens which exist in the Element ecosystem. This makes sense to have but it means that this tokenlist also includes a number of tokens which are unnecessary on a trading UI.

For example, users are discouraged from directly depositing into the `YVaultAssetProxy` contract so there's no reason for them to hold these tokens. Their inclusion just results in clutter in the token selection screen.

The ideal situation for us would be a sub-list which just includes the tokens with the tags "eP", "eY" and "underlying". The tokenlist generation script could be refactored to generate each sublist separately and then merge them into what exists today.

I understand that we could pull the existing tokenlist and then filter in the frontend however that needs special case behaviour for these particular assets which isn't ideal.

**Comment by DannyDelott** [comment]
This is an interesting problem. I think any app that wants to integrate the Element tokenlist will ultimately use just a subset of the included tokens, eg:

- (Balancer's trading UI case) depends on the `eP`, `eY`, and `underlying`, though a 3rd-party trading UI might need the `ccpool` and `wpool` tokens as well.
- A stats dashboard showing TVL across defi platforms (eg: defipulse) will depend on the `eP`, `ccpool`, `wpool`, and `assetproxy` tokens (to include the accumulated interest).
- An app that makes it easy to buy fixed yield (principal tokens only) will depend on the `ccpool` and `eP` tokens.

I'm a little wary of publishing a tokenlist for individual use-cases when the full list can filtered as you mention, though if depending on just the `eP`, `eY`, and `underlying` tokens ends up being a common use-case I'm open to it.

**Comment by TomAFrench** [comment]
Yeah, I can sympathise with ending up in a `2^n` tokenlist hell.

We could potentially perform the filtering ourselves and rehost the tokenlist. The downside to that is we'd need to make sure to keep it in sync with yours whereas the main usecase I see for it relies on us having the most up-to-date `eP` tokens to allow your users to roll over easily (and so relies on us promptly adding the new tokens).

**Comment by DannyDelott** [comment]
> We could potentially perform the filtering ourselves and rehost the tokenlist.

One of our goals is to publish this as an npm package, (eg: https://github.com/sambacha/yearn-finance-tokenlist) so it can be included in any app bundle.

Is your pattern to request the tokenlist.json from a url at runtime instead? Perhaps it's worth exploring a simple tokenlist server w/ a `tags` query param, eg: 

`http://element.fi/tokenlist.json?tags=eP,eY,underlying`

**Comment by TomAFrench** [comment]
> Is your pattern to request the tokenlist.json from a url at runtime instead?

Yes that's correct. We maintain a mapping of tokenlist urls which we query and merge at runtime to simplify updates.

> Perhaps it's worth exploring a simple tokenlist server w/ a `tags` query param, eg:
> 
> `http://element.fi/tokenlist.json?tags=eP,eY,underlying`

Yeah, this would work for us I think. :smiley:

**Comment by DannyDelott** [comment]
> > Perhaps it's worth exploring a simple tokenlist server w/ a `tags` query param, eg:
> > `http://element.fi/tokenlist.json?tags=eP,eY,underlying`
> 
> Yeah, this would work for us I think. 😃


On second thought, we might be able to do this quicker and without a server using just an AWS Lambda function. I'm not sure I can give a timeline on implementing this though.
